src/Deposit.js

```diff
@@ -978,10 +978,26 @@ export default class Deposit {
     console.debug(
       `Waiting for deposit ${this.address} to retrieve public key...`
     )
-    // Ask the deposit to fetch and store the signer pubkey.
-    const pubkeyTransaction = await EthereumHelpers.sendSafely(
-      this.contract.methods.retrieveSignerPubkey()
-    )
+
+    let pubkeyTransaction
+
+    for (;;) {
```

Comment:
I don't know about retrying infinitely here… It seems like a bit much, no? This could be a legitimate failure, after all.

I'm wondering if it might make sense to build the retry logic into `sendSafely`, with an optional parameter to specify the number of retries and an exponential backoff on the retry time. Thoughts?

Reply:
> I don't know about retrying infinitely here… It seems like a bit much, no? This could be a legitimate failure, after all.

Yeah, you're right.

> I'm wondering if it might make sense to build the retry logic into sendSafely, with an optional parameter to specify the number of retries and an exponential backoff on the retry time. Thoughts?

Yeah sure, we can make it generic.